schematic/store/synapse.py

```diff
@@ -1537,14 +1540,21 @@ async def format_row_annotations(
                 annos.pop(anno_k) if anno_k in annos.keys() else annos
             # Otherwise save annotation as approrpriate
             else:
+                if annotation_keys == "display_label":
```

Comment:
You may have picked this up as you've gone on your writing test journey, is that one big aspect of writing code is to write the code in a way that is more easily testable.  

There are probably some strategies to make these parts of code testable or at least this logic you have added.